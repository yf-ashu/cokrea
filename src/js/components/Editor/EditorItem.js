import React from 'react';
import PropTypes from 'prop-types';

const EditorItem = ({
    tag,
    attribute,
    option,
    style,
    outside,
    onMouseDown,
    onBlur,
    onDoubleClick,textContent
}) => {
    let Tag = tag;
    if (tag !== 'img') {
        option.dangerouslySetInnerHTML = { __html: textContent };
    }

    return (
        <div
            className="editorMain__items"
            style={outside}
            data-id={attribute.id}
            draggable="false"
        >
            <Tag
                onDoubleClick={onDoubleClick}
                data-id={attribute.id}
                className={attribute.className}
                id={attribute.id}
                type={attribute.type}
                data-format={attribute.format}
                style={style}
                suppressContentEditableWarning
                src={attribute.src}
                onBlur={onBlur}
                draggable="false"
                onMouseDown={onMouseDown}
                tabIndex="-1"
                {...option}
            >
            </Tag>
        </div>
    );
};

EditorItem.propTypes = {
    tag: PropTypes.string.isRequired,
    attribute: PropTypes.any.isRequired,
    option: PropTypes.any,
    style: PropTypes.any,
    outside: PropTypes.any,
    onBlur: PropTypes.func,
    onMouseDown: PropTypes.func,
    onDoubleClick: PropTypes.func,
    textContent:PropTypes.string,

};
export default EditorItem;
