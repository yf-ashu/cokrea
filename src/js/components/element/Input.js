import React from 'react';
import PropTypes from 'prop-types';
const Input = ({ type, id, value, handleChange,src }) => (
    <div className="input">
        <img className="input__icon" src={src} />
        <input
            type={type}
            className="input-control"
            id={id}
            value={value}
            onChange={handleChange}
        />
    </div>
);
Input.propTypes = {
    type: PropTypes.string,
    id: PropTypes.string,
    value: PropTypes.string,
    src: PropTypes.string,
    handleChange: PropTypes.func
};
export default Input;
