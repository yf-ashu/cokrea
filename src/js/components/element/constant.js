import phone from '../../../img/phone.svg';
import mail from '../../../img/mail.svg';
import h1 from '../../../img/text-01.png';
import h2 from '../../../img/text-02.png';
import p from '../../../img/text-04.png';
import span from '../../../img/text-03.png';

export const constant = {
    buttonDisplay: {
        text: [
            {
                type: 'h1',
                src: h1,
                size: { width: 0.8, height: 75 },
                textContent: 'Double click to edit',
                format: ['text', 'text']
            },
            {
                type: 'h2',
                src: h2,
                size: { width: 0.8, height: 55 },
                textContent: 'Double click to edit',
                format: ['text', 'text']
            },
            {
                type: 'p',
                src: p,
                size: { width: 0.8, height: 65 },
                format: ['text', 'text'],
                textContent:
                    'Double click to edit'
            },
            {
                type: 'span',
                src: span,
                size: { width: 0.8, height: 65 },
                format: ['text', 'text'],
                textContent:
                    'Double click to edit. This is a span text area.'
            }
        ],
        img: [
            {
                type: 'img',
                src: phone,
                size: { width: 0.3, height: 55 },
                format: ['image', 'phone']
            },
            {
                type: 'img',
                src: mail,
                size: { width: 0.3, height: 55 },
                format: ['image', 'mail']
            }
        ],
        square: []
    },
    color : ['red', 'yellow', 'orange', 'blue', 'purple', 'green'],
    editor:{
        display: [], //控制中間元素出現
        type: { type: null, name: null }, //控制左邊按鈕是否顯示
        buttonItem: [], //控制左邊預設按鈕出現
        editMainStyle: [
            {
                scale: 1,
                style: [
                    { width: 800 },
                    { height: 2000 },
                    { transform: 'scale(1)' },
                    { transformOrigin: '0 0' },
                    { backgroundColor: '#ffffff' }
                ]
            }
        ], //主畫布的style
        controllCurrent: ['page', null, null, null], //第一個是種類，第二個是主要的，第三個是移圖層用的
        fileUpload: { imgUrl: null, file: null },
        mouseEvent: 'true',
        saveButton: false,
        shareButton: [false, false],
        history: [],
        redoItem: [],
        loginStatus: null,
        projectData: null,
        database: null,
        othercontrollCurrent: [],
        loading: true,
        downloadUrl: null,
        intervalId: null,
        saved: [false, '已存檔'],
        trashCan: {},
        locationHref: null
    }

};

export const random = () => {
    let date = Date.now().toString();
    let possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_';
    let random = '';
    for (let i = 1; i < 15; i += 3) {
        let str = possible.charAt(Math.floor(Math.random() * possible.length));

        if (i === 1) {
            random =
                str +
                date.slice(0, i + 1) +
                str +
                date.slice(i + 2, date.length);
        } else {
            random =
                random.slice(0, i + 1) +
                str +
                random.slice(i + 2, random.length);
        }
    }
    return random;
};

export const connectFetch = (target, payload, fn) => {
    // console.log(payload)
    let url = `https://cokrea-editor.firebaseapp.com${target}`;
    // console.log(url)
    fetch(url, payload)
        .then(function(response) {
            return response.json();
        })
        .then(jsonData => {
            console.log(jsonData);
            return fn(jsonData);
        })
        .catch(err => {
            console.log('錯誤:', err);
            return fn(err);
        });
};

export const styleSetting = type => {
    let tmp = [];
    switch (type.type) {
    case 'img': {
        tmp = [{ backgroundColor: 'rgba(0,0,0,0)' },
            { color: '#000000' },
            {borderRadius:'0'}
        ];
        break;
    }
    case 'h1': {
        tmp = [
            { backgroundColor: 'rgba(0,0,0,0)' },
            { color: '#000000' },
            { fontWeight: 700 },
            { fontStyle: 'normal' },
            { fontSize: 47 }
        ];
        break;
    }
    case 'h2': {
        tmp = [
            { backgroundColor: 'rgba(0,0,0,0)' },
            { color: '#000000' },
            { fontWeight: 700 },
            { fontStyle: 'normal' },
            { fontSize: 36 }
        ];
        break;
    }
    case 'p': {
        tmp = [
            { backgroundColor: 'rgba(0,0,0,0)' },
            { color: '#000000' },

            { fontWeight: 700 },
            { fontStyle: 'normal' },
            { fontSize: 24 }
        ];
        break;
    }
    case 'span': {
        tmp = [
            { backgroundColor: 'rgba(0,0,0,0)' },
            { color: '#000000' },

            { fontWeight: 400 },
            { fontStyle: 'normal' },
            { fontSize: 16 }
        ];
        break;
    }
    default: {
        tmp = [
            { backgroundColor: 'rgba(0,0,0,0)' },
            { color: '#000000' },

            { fontWeight: 400 },
            { fontStyle: 'normal' },
            { fontSize: 16 }
        ];
    }
    }
    return tmp;
};
