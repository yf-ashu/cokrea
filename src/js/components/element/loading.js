import React from 'react';
import PropTypes from 'prop-types';

const loading = ({loading}) => {
    return <div className={loading?'loading':'loading displayOpacity'} />;
};

loading.propTypes = {
    loading:PropTypes.bool
};
export default loading;
