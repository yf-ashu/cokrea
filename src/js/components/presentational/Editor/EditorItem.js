import React from 'react';
import PropTypes from 'prop-types';

const EditorItem = ({
    tag,
    attribute,
    textContent,
    style,
    onClick,
    outside,
    contentEditable,
    onMouseDown,onBlur,onDoubleClick
}) => {
    let Tag = tag;
    return (
        <div
            className="editorMain__items"
            style={outside}
            data-id={attribute.id}
            onMouseDown={onMouseDown}
            onBlur={onBlur}
            onDoubleClick={onDoubleClick}
        >
            <Tag
                className={attribute.className}
                id={attribute.id}
                type={attribute.type}
                data-format={attribute.format}
                style={style}
                onClick={onClick}
                contentEditable={contentEditable}
                suppressContentEditableWarning
                src={attribute.src}
                // onBlur={onBlur}

            >
                {textContent}
            </Tag>
        </div>
    );
};

EditorItem.propTypes = {
    tag: PropTypes.string.isRequired,
    attribute: PropTypes.any.isRequired,
    textContent: PropTypes.string,
    style: PropTypes.any,
    outside: PropTypes.any,
    onClick: PropTypes.func,
    onBlur: PropTypes.func,
    contentEditable: PropTypes.string,
    onMouseDown: PropTypes.func,
    onDoubleClick:PropTypes.func
};
export default EditorItem;
