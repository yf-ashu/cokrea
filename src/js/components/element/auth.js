import { connectFetch } from './constant';
import firebase from 'firebase/app';
require('firebase/auth');
require('firebase/database');
require('firebase/storage');
export const authInfomation = func => {
    let storage, database;
    if (!firebase.apps.length) {
        let connect = initFirebase();
        database = connect.database();
        storage = connect.storage();
    }
    database = firebase.database();
    storage = firebase.storage();

    firebase.auth().onAuthStateChanged(user => {
        let authInfomation = [];
        if (user) {
            console.log('有登入');
            let data = {
                id: user.uid
            };
            console.log(user);
            let target = '/app/getAccount';
            let payload = {
                method: 'POST',
                body: JSON.stringify(data),
                headers: new Headers({
                    'Content-Type': 'application/json'
                })
            };
            let getMemberData = data => {
                let imgUrl = {};
                if (data.project) {
                    data.project.map(projectdata => {
                        console.log(projectdata.projectId);
                        storage
                            .ref(projectdata.projectId + '/canvas.png')
                            .getDownloadURL()
                            .then(url => {
                                imgUrl[projectdata.projectId] = url;
                                if (
                                    Object.keys(imgUrl).length ===
                                    data.project.length
                                ) {
                                    authInfomation = [
                                        user,
                                        data,
                                        [database, storage],
                                        imgUrl
                                    ];
                                    console.log(authInfomation);
                                    return func(authInfomation);
                                }
                            })
                            .catch(error => {
                                console.log(error);
                            });
                    });
                } else {
                    console.log('沒有圖片');
                    authInfomation = [user, data, [database, storage], null];
                    return func(authInfomation);
                }
            };
            console.log('在這裡');
            connectFetch(target, payload, getMemberData);
        } else {
            authInfomation = null;
            console.log('沒有登入');
            return func(authInfomation);
        }
    });
};
export const initFirebase = () => {
    let config = {
        apiKey: 'AIzaSyAncjAic0clz2IUCF-HOjHVOCe9_YCRzdo',
        authDomain: 'cokrea-editor.firebaseapp.com',
        databaseURL: 'https://cokrea-editor.firebaseio.com',
        projectId: 'cokrea-editor',
        storageBucket: 'cokrea-editor.appspot.com',
        messagingSenderId: '751312443170'
    };
    let firebaseConnect = firebase.initializeApp(config);
    var database = firebase.database();
    console.log(database);
    return firebaseConnect;
};
export const getAddress = () => {
    let projectDatahref = location.href.split('edit/')[1];
    return projectDatahref;
};

export const firebaseSet = (prePath,nextPath,value) => {
    let address=getAddress();

    firebase
        .database()
        .ref(prePath+'/' + address + '/'+nextPath)
        .set(value);
};

export const firebaseOnsite=(that,user)=>{
    let database=firebase.database();
    let address=getAddress();

    const connectedRef = database.ref('.info/connected');
    connectedRef.off();
    connectedRef.on('value', function(snap) {
        if (snap.val() === true) {
            let dataconnect = database.ref(
                'temp/' + address+ '/nowclick/' + user.uid
            );
            dataconnect.set([that.state.controllCurrent, user.displayName]);
            dataconnect.onDisconnect().remove();
            database
                .ref('temp/' + address + '/nowclick')
                .on('value', snapshot => {
                    that.setState({
                        othercontrollCurrent: snapshot.val()
                            ? snapshot.val()
                            : []
                    });
                });

            let trash = database.ref('temp/' + address + '/trash');
            trash.onDisconnect().remove();
            database
                .ref('temp/' + address + '/trash')
                .on('value', snapshot => {
                    if (snapshot.val()) {
                        let combine;
                        if (that.state.trashCan !== null) {
                            combine = Object.assign(
                                snapshot.val(),
                                that.state.trashCan
                            );
                        }
                        that.setState({
                            trashCan: combine
                        });
                    }
                });

            that.setState({
                loading: false
            });
        } else {
            console.log('離線');
            firebaseSet('temp', 'nowclick/' + user.uid, 'null');
            firebaseSet('temp', 'trash', that.state.trashCan);
        }
    });
};
export const firebaseGetProjectData=(that)=>{
    let database=firebase.database();
    let address=getAddress();
    database.ref('/projectData/' + address).off();
    database
        .ref('/projectData/' + address)
        .once('value', snapshot => {
            if (snapshot.exists()) {
                let data = snapshot.val();
                if (snapshot.val().owner === that.state.loginStatus.uid) {
                    let shareButton = that.state.shareButton;
                    shareButton[1] = true;
                    that.setState(
                        {
                            shareButton: shareButton
                        },
                        () => {
                            that.connectDb(
                                database,
                                that.state.loginStatus
                            );
                        }
                    );
                } else if (data.share[1] !== 'no data') {
                    let okMail= data.share[1].filter(userdata => {
                        return  userdata === that.state.loginStatus.email;
                        
                    });
                    if (okMail.length===0) {
                        alert('沒有存取權');
                        window.location.pathname = '/';
                    }else{
                        that.connectDb(
                            database,
                            that.state.loginStatus
                        );
                    }
                } else {
                    if (data.share[0].public === 'public') {
                        window.location.pathname =
                            '/views/' + address;
                    } else {
                        alert('沒有存取權');
                        window.location.pathname = '/';
                    }
                }
            } else {
                alert('沒有此檔案');
                window.location.pathname = '/';
            }
        });


};