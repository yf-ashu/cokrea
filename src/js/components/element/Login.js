import React, { Component } from 'react';
import Input from './Input';
import user from '../../../img/user.svg';
import PropTypes from 'prop-types';
import firebase from 'firebase/app';
require('firebase/auth');
import { connectFetch } from './constant';
import {  authInfomation } from './auth';
import CreateHeader from '../Create/CreateHeader';
import Loading from './loading';
import { Redirect, NavLink } from 'react-router-dom';
import iconColor from '../../../img/icon-color.png';

class Login extends Component {
    constructor(props) {
        +super(props);
        this.state = {
            userName: '',
            password: '',
            database: null,
            storage: null,
            loading: true
        };
        this.signupWithEmail = this.signupWithEmail.bind(this);
        this.loginWithEmail = this.loginWithEmail.bind(this);
        this.loginWithFB = this.loginWithFB.bind(this);
        this.loginWithGoogle = this.loginWithGoogle.bind(this);
        this.handleInput = this.handleInput.bind(this);
    }
    componentDidMount() {
       
        let authCheck = data => {
            console.log(data);
            if (data) {
                this.props.getUserData(data[0], data[1], data[2], data[3]);
            } else {
                this.setState({
                    loading: false
                });
            }
        };

        authInfomation(authCheck);
    }

    handleInput(e, type) {
        this.setState({
            [type]: e.currentTarget.value
        });
    }
    addTodatabase(result){
        let data = {
            userEmail: result.user.providerData[0].email,
            userId: result.user.uid,
            providerId: result.user.providerData[0].providerId
        };
        let target = '/app/manageAccount';
        let payload = {
            method: 'POST',
            body: JSON.stringify(data),
            headers: new Headers({
                'Content-Type': 'application/json'
            })
        };
        let getMemberData = data => {
            console.log('有', data);
        };
        connectFetch(target, payload, getMemberData);
    }
    loginWithFB() {
        let that=this;
        let provider = new firebase.auth.FacebookAuthProvider();
        firebase
            .auth()
            .signInWithPopup(provider)
            .then(function(result) {
                that.addTodatabase(result);
            })
            .catch(function(error) {
                console.log(error);
            });
    }
    loginWithGoogle() {
        let that=this;

        let provider = new firebase.auth.GoogleAuthProvider();
        firebase
            .auth()
            .signInWithPopup(provider)
            .then(function(result) {
                that.addTodatabase(result);
            })
            .catch(function(error) {
                console.log(error);
            });
    }
    loginWithEmail() {
        let email = this.state.userName;
        let password = this.state.password;
        if (!email || !password) return alert('請輸入');
        let mailForm = /^[\w]+@[\w_]+(\.\w{2,4})/;
        if (!mailForm.test(email)) {
            return alert('格式錯誤');
        }

        firebase
            .auth()
            .signInWithEmailAndPassword(email, password)
            .then(function(result) {
                console.log('創完了');

                console.log(result);
            })
            .catch(function(error) {
                if (
                    error.code == 'auth/invalid-email' ||
                    error.code == 'auth/user-not-found'
                ) {
                    alert('查無此Email。');
                } else if (error.code == 'auth/wrong-password') {
                    alert('密碼錯誤，請重新輸入。');
                } else {
                    alert(error.message, false);
                }
                console.log(error);
            });
    }
    signupWithEmail() {
        let email = this.state.userName;
        let password = this.state.password;

        let mailForm = /^[\w]+@[\w_]+(\.\w{2,4})/;
        if (!email || !password) return alert('請輸入');
        if (!mailForm.test(email)) {
            alert('格式錯誤');
            return;
        }
        if (password.length < 7) {
            alert('請輸入密碼');
            return;
        }let that=this;
        firebase
            .auth()
            .createUserWithEmailAndPassword(email, password)
            .then(function(result)  {
                alert('註冊成功並自動登入');
                that.addTodatabase(result);
                console.log(result);
            })
            .catch(error => { 
                console.log('再觀察');
                console.log(error);
            });
    }
    render() {
        if (this.props.loginStatus && this.props.userData) {
            return <Redirect to="/dashboard" />;
        }
        let address = location.pathname.split('/')[1];
        console.log(address);
        return (
            <div className="login">
                <Loading loading={this.state.loading} />
                <CreateHeader
                    login={this.props.loginStatus}
                    logout={this.logout}
                    memberButton={this.state.memberButton}
                    handlememberButton={e => {
                        this.setState({
                            memberButton: e
                        });
                    }}
                />

                {address==='login'?(
                    <div className="loginMain">
                        <div className="login__main">
                            <div className="login__inner">
                                <img src={iconColor} />
                                <Input
                                    src={user}
                                    type="text"
                                    id="loginUser"
                                    className="input-login input-control"
                                    onChange={e => {
                                        this.handleInput(e, 'userName');
                                    }}
                                    value={this.state.userName}
                                    text="Email"
                                />
                                <Input
                                    src={user}
                                    type="password"
                                    id="loginPwd"
                                    className="input-login input-control"
                                    onChange={e => {
                                        this.handleInput(e, 'password');
                                    }}
                                    value={this.state.password}
                                    text="Password"
                                />
                                <div className="login__buttons">
                        
                                    <button
                                        className="login__button"
                                        onClick={this.loginWithEmail}
                                    >
                                        Login
                                    </button>
                                </div>
                                <div className="login__line" />
                                <button
                                    className="login__fb"
                                    onClick={this.loginWithFB}
                                >
                                    Login with Facebook
                                </button>
                                <button
                                    className="login__google"
                                    onClick={this.loginWithGoogle}
                                >
                                    Login with Google
                                </button>
                                <NavLink to="/signup" className="login__link">
                                    New here? Create an account.
                                </NavLink>
                            </div>
                        </div>
                    </div>):

                    (       <div className="loginMain">
                        <div className="login__main">
                            <div className="login__inner">
                                <img src={iconColor} />
                                <Input
                                    src={user}
                                    type="text"
                                    id="loginUser"
                                    className="input-login input-control"
                                    onChange={e => {
                                        this.handleInput(e, 'userName');
                                    }}
                                    value={this.state.userName}
                                    text="Email"
                                />
                                <Input
                                    src={user}
                                    type="password"
                                    id="loginPwd"
                                    className="input-login input-control"
                                    onChange={e => {
                                        this.handleInput(e, 'password');
                                    }}
                                    value={this.state.password}
                                    text="Password"
                                    placeholder="at least 6 characters."
                                />
                                <div className="login__buttons">
                            
                                    <button
                                        className="login__button"
                                        onClick={this.signupWithEmail}
                                    >
                                    Signup
                                    </button>
                                </div>
                           
                            </div>
                        </div>
                    </div>)
                }
            </div>
        );
    }
}
Login.propTypes = {
    getUserData: PropTypes.func.isRequired,
    loading: PropTypes.any,
    loginStatus: PropTypes.string,
    userData: PropTypes.string
};
export default Login;
