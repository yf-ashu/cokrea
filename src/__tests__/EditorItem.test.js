import React from 'react';
import EditorItem from '../js/components/Editor/EditorItem';
import sample from '../js/components/element/sample.json';
import {initFirebase} from '../js/components/element/auth';
import firebase from 'firebase/app';
require('firebase/database');
require('firebase/storage');
const Enzyme = require('enzyme');
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import { mount } from 'enzyme';

describe( 'EditorItem tests', () => {
    let smapleData=sample.sample.sample1.display[0];
    initFirebase();
    let database = firebase.database();

    it( 'data is acquired from firebase correctly', done => {
        database
            .ref('/projectData/testData')
            .on('value', snapshot => {
                // let data = snapshot.val();
              
                // console.log(data);

                done();
            });
    });


    let wrapper = mount( <EditorItem 
        key={smapleData.key}
        tag={smapleData.tag}
        attribute={smapleData.attribute}
        outside={Object.assign({}, ...smapleData.outside)}
        option={Object.assign({}, ...smapleData.option)}
        style={Object.assign({}, ...smapleData.style)}
        onMouseDown={jest.fn()}
        onBlur={jest.fn()}
        onDoubleClick={jest.fn()}
        textContent={smapleData.textContent}
    
    /> );
    wrapper.update();
    it( 'EditorItem is rendered correctly' , () => { 
        expect(wrapper).toMatchSnapshot();
    });
    it( 'EditorItem outside toHaveProperty' , () => { 
        expect(wrapper.props().outside).toHaveProperty('width' ,expect.any(Number));
        expect(wrapper.props().outside).toHaveProperty('height' ,expect.any(Number));
        expect(wrapper.props().outside).toHaveProperty('left' ,expect.any(Number));
        expect(wrapper.props().outside).toHaveProperty('top' ,expect.any(Number));
    });

});

