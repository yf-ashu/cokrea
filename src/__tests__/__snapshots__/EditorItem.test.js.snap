// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorItem tests EditorItem is rendered correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <EditorItem
    attribute={
      Object {
        "className": "editorMain__item X15X0R22O61g50x editorMain__item--h1",
        "format": "text,text",
        "id": "X15X0R22O61g50x",
        "src": "",
        "type": "h1",
      }
    }
    onBlur={[MockFunction]}
    onDoubleClick={[MockFunction]}
    onMouseDown={[MockFunction]}
    option={
      Object {
        "contentEditable": "false",
        "dangerouslySetInnerHTML": Object {
          "__html": "",
        },
      }
    }
    outside={
      Object {
        "height": 7,
        "left": 80.667,
        "top": 223.334,
        "width": 528,
      }
    }
    style={
      Object {
        "backgroundColor": "#ffd300",
        "color": "#000000",
        "fontSize": 47,
        "fontStyle": "normal",
        "fontWeight": 700,
        "height": "100%",
        "width": "100%",
      }
    }
    tag="h1"
    textContent=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "attribute": Object {
        "className": "editorMain__item X15X0R22O61g50x editorMain__item--h1",
        "format": "text,text",
        "id": "X15X0R22O61g50x",
        "src": "",
        "type": "h1",
      },
      "onBlur": [MockFunction],
      "onDoubleClick": [MockFunction],
      "onMouseDown": [MockFunction],
      "option": Object {
        "contentEditable": "false",
        "dangerouslySetInnerHTML": Object {
          "__html": "",
        },
      },
      "outside": Object {
        "height": 7,
        "left": 80.667,
        "top": 223.334,
        "width": 528,
      },
      "style": Object {
        "backgroundColor": "#ffd300",
        "color": "#000000",
        "fontSize": 47,
        "fontStyle": "normal",
        "fontWeight": 700,
        "height": "100%",
        "width": "100%",
      },
      "tag": "h1",
      "textContent": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="editorMain__items"
        data-id="X15X0R22O61g50x"
        draggable="false"
        style="width: 528px; height: 7px; left: 80.667px; top: 223.334px;"
      >
        <h1
          class="editorMain__item X15X0R22O61g50x editorMain__item--h1"
          contenteditable="false"
          data-format="text,text"
          data-id="X15X0R22O61g50x"
          draggable="false"
          id="X15X0R22O61g50x"
          src=""
          style="width: 100%; height: 100%; background-color: rgb(255, 211, 0); color: rgb(0, 0, 0); font-weight: 700; font-style: normal; font-size: 47px;"
          tabindex="-1"
          type="h1"
        />
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="editorMain__item X15X0R22O61g50x editorMain__item--h1"
          contentEditable="false"
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
          data-format="text,text"
          data-id="X15X0R22O61g50x"
          draggable="false"
          id="X15X0R22O61g50x"
          onBlur={[MockFunction]}
          onDoubleClick={[MockFunction]}
          onMouseDown={[MockFunction]}
          src=""
          style={
            Object {
              "backgroundColor": "#ffd300",
              "color": "#000000",
              "fontSize": 47,
              "fontStyle": "normal",
              "fontWeight": 700,
              "height": "100%",
              "width": "100%",
            }
          }
          suppressContentEditableWarning={true}
          tabIndex="-1"
          type="h1"
        />,
        "className": "editorMain__items",
        "data-id": "X15X0R22O61g50x",
        "draggable": "false",
        "style": Object {
          "height": 7,
          "left": 80.667,
          "top": 223.334,
          "width": 528,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h1
            class="editorMain__item X15X0R22O61g50x editorMain__item--h1"
            contenteditable="false"
            data-format="text,text"
            data-id="X15X0R22O61g50x"
            draggable="false"
            id="X15X0R22O61g50x"
            src=""
            style="width: 100%; height: 100%; background-color: rgb(255, 211, 0); color: rgb(0, 0, 0); font-weight: 700; font-style: normal; font-size: 47px;"
            tabindex="-1"
            type="h1"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "editorMain__item X15X0R22O61g50x editorMain__item--h1",
            "contentEditable": "false",
            "dangerouslySetInnerHTML": Object {
              "__html": "",
            },
            "data-format": "text,text",
            "data-id": "X15X0R22O61g50x",
            "draggable": "false",
            "id": "X15X0R22O61g50x",
            "onBlur": [MockFunction],
            "onDoubleClick": [MockFunction],
            "onMouseDown": [MockFunction],
            "src": "",
            "style": Object {
              "backgroundColor": "#ffd300",
              "color": "#000000",
              "fontSize": 47,
              "fontStyle": "normal",
              "fontWeight": 700,
              "height": "100%",
              "width": "100%",
            },
            "suppressContentEditableWarning": true,
            "tabIndex": "-1",
            "type": "h1",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "h1",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "attribute": Object {
          "className": "editorMain__item X15X0R22O61g50x editorMain__item--h1",
          "format": "text,text",
          "id": "X15X0R22O61g50x",
          "src": "",
          "type": "h1",
        },
        "onBlur": [MockFunction],
        "onDoubleClick": [MockFunction],
        "onMouseDown": [MockFunction],
        "option": Object {
          "contentEditable": "false",
          "dangerouslySetInnerHTML": Object {
            "__html": "",
          },
        },
        "outside": Object {
          "height": 7,
          "left": 80.667,
          "top": 223.334,
          "width": 528,
        },
        "style": Object {
          "backgroundColor": "#ffd300",
          "color": "#000000",
          "fontSize": 47,
          "fontStyle": "normal",
          "fontWeight": 700,
          "height": "100%",
          "width": "100%",
        },
        "tag": "h1",
        "textContent": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="editorMain__items"
          data-id="X15X0R22O61g50x"
          draggable="false"
          style="width: 528px; height: 7px; left: 80.667px; top: 223.334px;"
        >
          <h1
            class="editorMain__item X15X0R22O61g50x editorMain__item--h1"
            contenteditable="false"
            data-format="text,text"
            data-id="X15X0R22O61g50x"
            draggable="false"
            id="X15X0R22O61g50x"
            src=""
            style="width: 100%; height: 100%; background-color: rgb(255, 211, 0); color: rgb(0, 0, 0); font-weight: 700; font-style: normal; font-size: 47px;"
            tabindex="-1"
            type="h1"
          />
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="editorMain__item X15X0R22O61g50x editorMain__item--h1"
            contentEditable="false"
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            data-format="text,text"
            data-id="X15X0R22O61g50x"
            draggable="false"
            id="X15X0R22O61g50x"
            onBlur={[MockFunction]}
            onDoubleClick={[MockFunction]}
            onMouseDown={[MockFunction]}
            src=""
            style={
              Object {
                "backgroundColor": "#ffd300",
                "color": "#000000",
                "fontSize": 47,
                "fontStyle": "normal",
                "fontWeight": 700,
                "height": "100%",
                "width": "100%",
              }
            }
            suppressContentEditableWarning={true}
            tabIndex="-1"
            type="h1"
          />,
          "className": "editorMain__items",
          "data-id": "X15X0R22O61g50x",
          "draggable": "false",
          "style": Object {
            "height": 7,
            "left": 80.667,
            "top": 223.334,
            "width": 528,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h1
              class="editorMain__item X15X0R22O61g50x editorMain__item--h1"
              contenteditable="false"
              data-format="text,text"
              data-id="X15X0R22O61g50x"
              draggable="false"
              id="X15X0R22O61g50x"
              src=""
              style="width: 100%; height: 100%; background-color: rgb(255, 211, 0); color: rgb(0, 0, 0); font-weight: 700; font-style: normal; font-size: 47px;"
              tabindex="-1"
              type="h1"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "editorMain__item X15X0R22O61g50x editorMain__item--h1",
              "contentEditable": "false",
              "dangerouslySetInnerHTML": Object {
                "__html": "",
              },
              "data-format": "text,text",
              "data-id": "X15X0R22O61g50x",
              "draggable": "false",
              "id": "X15X0R22O61g50x",
              "onBlur": [MockFunction],
              "onDoubleClick": [MockFunction],
              "onMouseDown": [MockFunction],
              "src": "",
              "style": Object {
                "backgroundColor": "#ffd300",
                "color": "#000000",
                "fontSize": 47,
                "fontStyle": "normal",
                "fontWeight": 700,
                "height": "100%",
                "width": "100%",
              },
              "suppressContentEditableWarning": true,
              "tabIndex": "-1",
              "type": "h1",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "h1",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
